drivers/clock_control/clock_control_nrf2_common.c

```diff
@@ -159,3 +167,37 @@ int api_nosys_on_off(const struct device *dev, clock_control_subsys_t sys)
 
 	return -ENOSYS;
 }
+
+static void sync_cb(struct onoff_manager *mgr, struct onoff_client *cli, uint32_t state, int res)
+{
+	struct clock_control_nrf_sync_req_data  *data =
+		CONTAINER_OF(cli, struct clock_control_nrf_sync_req_data, cli);
+
+	data->res = res;
+	k_sem_give(data->sem);
+}
+
+int nrf_clock_control_request_sync(const struct device *dev, const struct nrf_clock_spec *spec)
+{
+	struct clock_control_nrf_sync_req_data data;
+	struct k_sem sem;
+	int err;
+
+	__ASSERT_NO_MSG(!k_is_in_isr());
+
+	k_sem_init(&sem, 0, 1);
+	data.sem = &sem;
```

Comment:
Why not make this semaphore a member of this structure? And initialize it statically?